### PR TITLE
for testing purposes only

### PR DESCRIPTION
**Pull request by MadWasp79**

### CODE REVIEW

File: native/src/binaryRead.cpp [hunk 1]
```diff
@@ -1039,7 +1132,678 @@ MapDataObject* readMapDataObject(CodedInputStream* input, MapTreeBounds* tree, S
...
+				y += ctop;
+				break;
+			}
+			default: {
+				if (WireFormatLite::GetTagWireType(tag) == WireFormatLite::WIRETYPE_END_GROUP) {
+				// if (dataObject->getName("en").length() == 0) {
+				// 	dataObject->enName(TransliterationHelper.transliterate(dataObject.getName()));
+				// }
+					if (x != 0 || y != 0) {
+                        exit->setLocation(TRANSPORT_STOP_ZOOM, x, y);
```
Comment (vshcherb) [inline native/src/binaryRead.cpp:1175]: indent

File: native/src/binaryRead.cpp [hunk 2]
```diff
@@ -2230,5 +3012,11 @@ BinaryMapFile* initBinaryMapFile(std::string inputName, bool useLive, bool routi
 	}
 	
 	openFiles.push_back(mapFile);
+    transportIndexesList.insert(transportIndexesList.end(), mapFile->transportIndexes.begin(), mapFile->transportIndexes.end());
+    
 	return mapFile;
 }
+
+std::vector<BinaryMapFile* > getOpenMapFiles() {
```
Comment (vshcherb) [inline native/src/binaryRead.cpp:3020]: suspicious ???

File: native/src/transportRoutingContext.h [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+                        loadedWays++;
+                        res.push_back(it);
+                    }
+                }
+            }
+        }
+    }
+    
+    SearchQuery* buildSearchTransportRequest(int sleft, int sright, int stop, int sbottom, int limit, vector<SHARED_PTR<TransportStop>>& stops)
+    {
```
Comment (vshcherb) [inline native/src/transportRoutingContext.h:106]: java-style of {}

File: native/src/generalRouter.h [hunk 4]
```diff
@@ -274,6 +277,26 @@ class RouteAttributeContext {
     SHARED_PTR<RouteAttributeEvalRule> getLastRule() {
         return rules.back();
     }
+    
+    int evaluateInt(dynbitset& rawTypes, int defValue)
```
Comment (vshcherb) [inline native/src/generalRouter.h:281]: java-style {}

File: native/src/transportRoutePlanner.cpp [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+            }
+            if(includeRoute(s, route)) {
+                include = true;
+                route->to_string();
+                break;
+            }
+        }
+        if(!include) {
+            lst.push_back(route);
+            // System.out.println(route.toString());
```
Comment (vshcherb) [inline native/src/transportRoutePlanner.cpp:108]: ?

File: native/src/transportRoutePlanner.cpp [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+		//updateCalculationProgress(ctx, queue);
+    }
+	pt_timer.Pause();
+	OsmAnd::LogPrintf(OsmAnd::LogSeverityLevel::Info, "[NATIVE] PT calculation took %.3f ms, loading tiles (overall): %.3f ms, readTime : %.3f ms", 
+	pt_timer.GetElapsedMs(), ctx->loadTime.GetElapsedMs(), ctx->readTime.GetElapsedMs());
+
+    return prepareResults(ctx, results);
+}
+
+// private void initProgressBar(TransportRoutingContext ctx, LatLon start, LatLon end) {
```
Comment (vshcherb) [inline native/src/transportRoutePlanner.cpp:284]: ???

File: native/src/transportRoutePlanner.h [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+typedef priority_queue<SHARED_PTR<TransportRouteSegment>, vector<SHARED_PTR<TransportRouteSegment>>, TransportSegmentsComparator> TRANSPORT_SEGMENTS_QUEUE;
+
+class TransportRoutePlanner
+{
+public:
+    TransportRoutePlanner();
+    ~TransportRoutePlanner();
+    
+    vector<SHARED_PTR<TransportRouteResult>> buildTransportRoute(SHARED_PTR<TransportRoutingContext>& ctx);
```
Comment (vshcherb) [inline native/src/transportRoutePlanner.h:31]: TransportRoutingContext*

File: native/src/transportRouteResult.h [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+#include "transportRoutingContext.h"
+ 
+
+struct TransportRouteResult {
+    vector<SHARED_PTR<TransportRouteResultSegment>> segments;
+    double finishWalkDist;
+    double routeTime;
+    SHARED_PTR<TransportRoutingConfiguration> config;
+
+    TransportRouteResult(TransportRoutingContext* ctx) {
```
Comment (vshcherb) [inline native/src/transportRouteResult.h:14]: ctx -reference ??

Comment (vshcherb) [reply native/src/transportRouteResult.h:14]: TransportRoutingConfiguration, TransportRoutingContext should not be shared_ptr

Comment (vshcherb) [reply native/src/transportRouteResult.h:14]: vector<SHARED_PTR<TransportRouteResultSegment>> segments; - shared_ptr ? should be vector<TransportRouteResultSegment>

File: native/src/transportRouteResultSegment.h [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+    private:
+        static const bool DISPLAY_FULL_SEGMENT_ROUTE = false;
+        static const int  DISPLAY_SEGMENT_IND = 0;
+
+    public:
+    
+        TransportRouteResultSegment() {
+        }
+    
+        SHARED_PTR<TransportRoute> route;
```
Comment (vshcherb) [inline native/src/transportRouteResultSegment.h:18]: really needed?

File: native/src/transportRouteResultSegment.h [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+            double minStart = 150;
+            double minEnd = 150;
+            
+            double startLat = getStart()->lat;
+            double startLon = getStart()->lon;
+            double endLat = getEnd()->lat;
+            double endLon = getEnd()->lon;
+            
+            int endInd = -1;
+            vector<SHARED_PTR<Node>> res;
```
Comment (vshcherb) [inline native/src/transportRouteResultSegment.h:91]: Node, Way should not be shared_ptr

File: native/src/transportRoutingConfiguration.h [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+            walkChangeRadius =  router->getIntAttribute("walkChangeRadius", walkChangeRadius);
+            zoomToLoadTiles =  router->getIntAttribute("zoomToLoadTiles", zoomToLoadTiles);
+            maxNumberOfChanges =  router->getIntAttribute("maxNumberOfChanges", maxNumberOfChanges);
+            maxRouteTime =  router->getIntAttribute("maxRouteTime", maxRouteTime);
+            finishTimeSeconds =  router->getIntAttribute("delayForAlternativesRoutes", finishTimeSeconds);
+            string mn = router->getAttribute("max_num_changes");
+            int maxNumOfChanges = 3;
+            try
+            {
+                maxNumOfChanges = stoi(mn);
```
Comment (vshcherb) [inline native/src/transportRoutingConfiguration.h:47]: atoi ... parse number - method already exists

File: native/src/transportRoutingContext.h [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+struct TransportRouteSegment;
+struct TransportStop;
+struct TransportRoute;
+
+struct TransportRoutingContext
+{
+
+	SHARED_PTR<RouteCalculationProgress> calculationProgress;
+	UNORDERED(map)<int64_t, SHARED_PTR<TransportRouteSegment>> visitedSegments;
+	SHARED_PTR<TransportRoutingConfiguration> cfg;
```
Comment (vshcherb) [inline native/src/transportRoutingContext.h:21]: Remove SHARED_PTR

File: native/src/transportRouteResultSegment.cpp [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+    double minStart = 150;
+    double minEnd = 150;
+
+    double startLat = getStart()->lat;
+    double startLon = getStart()->lon;
+    double endLat = getEnd()->lat;
+    double endLon = getEnd()->lon;
+
+    int endInd = -1;
+    vector<SHARED_PTR<Node>> res;
```
Comment (vshcherb) [inline native/src/transportRouteResultSegment.cpp:73]: Remove SHARED_PTR

File: native/src/transportRouteResultSegment.cpp [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+    if (DISPLAY_FULL_SEGMENT_ROUTE)
+    {
+        if (route->forwardWays.size() > DISPLAY_SEGMENT_IND)
+        {
+            list.push_back(route->forwardWays[DISPLAY_SEGMENT_IND]);
+            return list;
+        }
+        return route->forwardWays;
+    }
+    vector<SHARED_PTR<Way>> fw = route->forwardWays;
```
Comment (vshcherb) [inline native/src/transportRouteResultSegment.cpp:63]: Remove SHARED_PTR

File: native/src/transportRouteSegment.h [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+
+    bool wasVisited(SHARED_PTR<TransportRouteSegment>& rrs);
+    SHARED_PTR<TransportStop> getStop(int i);
+    pair<double, double> getLocation();
+    double getLocationLat();
+    double getLocationLon();
+    int32_t getLength();
+    int64_t getId();
+    int32_t getDepth();
+    string to_string();
```
Comment (vshcherb) [inline native/src/transportRouteSegment.h:44]: _ name ? toString() to follow name conventions

File: native/src/transportRoutingConfiguration.cpp [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+		walkChangeRadius = router->getIntAttribute("walkChangeRadius", walkChangeRadius);
+		zoomToLoadTiles = router->getIntAttribute("zoomToLoadTiles", zoomToLoadTiles);
+		maxNumberOfChanges = router->getIntAttribute("maxNumberOfChanges", maxNumberOfChanges);
+		maxRouteTime = router->getIntAttribute("maxRouteTime", maxRouteTime);
+		finishTimeSeconds = router->getIntAttribute("delayForAlternativesRoutes", finishTimeSeconds);
+		string mn = router->getAttribute("max_num_changes");
+		int maxNumOfChanges = 3;
+		try
+		{
+			maxNumOfChanges = stoi(mn);
```
Comment (vshcherb) [inline native/src/transportRoutingConfiguration.cpp:24]: stoi ??? look parselength

File: native/src/transportRoutingConfiguration.cpp [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+		walkSpeed = spds.evaluateFloat(bs, walkSpeed);
+	}
+}
+
+float TransportRoutingConfiguration::getSpeedByRouteType(std::string routeType)
+{
+	const auto it = speed.find(routeType);
+	float sl = defaultTravelSpeed;
+	if (it == speed.end())
+	{
```
Comment (vshcherb) [inline native/src/transportRoutingConfiguration.cpp:54]: Java style {}

File: native/src/transportRouteResult.h [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+#include "CommonCollections.h"
+#include "commonOsmAndCore.h"
+
+struct TransportRouteResultSegment;
+struct TransportRoutingConfiguration;
+struct TransportRoutingContext;
+
+struct TransportRouteResult {
+
+    vector<SHARED_PTR<TransportRouteResultSegment>> segments;
```
Comment (vshcherb) [inline native/src/transportRouteResult.h:12]: unique_ptr

File: native/src/transportRouteResultSegment.h [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+	int32_t end;
+	double walkDist;
+	int32_t depTime;
+
+	TransportRouteResultSegment();
+
+	int getArrivalTime();
+	double getTravelDist();
+	vector<SHARED_PTR<Way>> getGeometry();
+	SHARED_PTR<TransportStop> getStart();
```
Comment (vshcherb) [inline native/src/transportRouteResultSegment.h:34]: const TransportStop&

File: native/src/transportRoutingConfiguration.cpp (hunk 16 above)
Comment (MadWasp79) [reply native/src/transportRoutingConfiguration.cpp:24]: done

File: native/src/transportRouteResultSegment.cpp (hunk 13 above)
Comment (MadWasp79) [reply native/src/transportRouteResultSegment.cpp:73]: done

File: native/src/transportRoutingConfiguration.h (hunk 11 above)
Comment (MadWasp79) [reply native/src/transportRoutingConfiguration.h]: done

File: native/src/transportRouteResultSegment.h (hunk 10 above)
Comment (MadWasp79) [reply native/src/transportRouteResultSegment.h]: done

File: native/src/transportRouteResultSegment.cpp (hunk 14 above)
Comment (MadWasp79) [reply native/src/transportRouteResultSegment.cpp:63]: done

File: native/src/transportRouteSegment.h (hunk 15 above)
Comment (MadWasp79) [reply native/src/transportRouteSegment.h:44]: done

File: native/src/transportRoutingConfiguration.cpp (hunk 17 above)
Comment (MadWasp79) [reply native/src/transportRoutingConfiguration.cpp:54]: done

File: native/src/binaryRead.cpp (hunk 1 above)
Comment (MadWasp79) [reply native/src/binaryRead.cpp:1175]: done

File: native/src/binaryRead.cpp (hunk 2 above)
Comment (MadWasp79) [reply native/src/binaryRead.cpp:3020]: its okay

File: native/src/transportRoutingContext.h (hunk 12 above)
Comment (MadWasp79) [reply native/src/transportRoutingContext.h:21]: changed to unique_ptr

File: native/src/transportRouteResultSegment.h (hunk 19 above)
Comment (MadWasp79) [reply native/src/transportRouteResultSegment.h:34]: done

File: native/src/transportRouteResultSegment.h (hunk 9 above)
Comment (MadWasp79) [reply native/src/transportRouteResultSegment.h]: seems like yes. Routes, Stops - used in several objects.

File: native/src/transportRouteResult.h (hunk 18 above)
Comment (MadWasp79) [reply native/src/transportRouteResult.h:12]: done

**Pull request closed without merging**